src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenDynamicTests.cs

```diff
@@ -54,7 +54,7 @@ public class CodeGen_DynamicTests : CSharpTestBase
             return (expectedUnoptimizedIL != null) ^ (expectedOptimizedIL != null) ? (unoptimizedVerifier ?? optimizedVerifier) : null;
         }
 
-        private readonly CSharpParseOptions _localFunctionParseOptions = TestOptions.Regular.WithLocalFunctionsFeature();
```

Comment:
Instead of going through every single test and removing these calls, could we just change implementation of the WithLocalFunctionsFeature and the like methods to assert that the options target CSharp7 or above?